evap/staff/views.py

```diff
@@ -790,12 +798,61 @@ def questionnaire_copy(request, questionnaire_id):
         else:
             return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
     else:
-        questionnaire = get_object_or_404(Questionnaire, id=questionnaire_id)
+        form, formset = get_identical_form_and_formset(questionnaire_id)
+        return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
+
+
+def find_questionnaire_name(old_questionnaire):
```

Comment:
i think it would be better to just have a version number and no dates at all. when creating a new version of "Questionnaire (01.01.2016)" it might become "Questionnaire (01.01.2016) (02.01.2016)" - that's not really helpful.

it would probably be the best if you could check whether a version number already exists and increment it when creating a new version. what do you think?
and please make the counter start at 2, creating a new version of "Questionnaire" should be named "Questionnaire (2)", not "Questionnaire (1)".